src/node_buffer.cc

```diff
@@ -965,11 +965,11 @@ void IndexOfString(const FunctionCallbackInfo<Value>& args) {
   size_t result = haystack_length;
 
   if (enc == UCS2) {
-    String::Value needle_value(isolate, needle);
-    if (*needle_value == nullptr)
-      return args.GetReturnValue().Set(-1);
+    std::unique_ptr<uint16_t[]> needle_buffer(new uint16_t[needle->Length()]);
```

Comment:
Why do you need a unique ptr here?

Reply:
I'm not a CPP pointers expert :sweat_smile:, I've removed the `unique_ptr`